components/omega/src/ocn/VertMix.h

```diff
@@ -0,0 +1,210 @@
+#ifndef OMEGA_VERTMIX_H
+#define OMEGA_VERTMIX_H
+//===-- ocn/VertMix.cpp - Vertical Mixing Coefficients -----------*- C++
+//-*-===//
+//
+/// \file
+/// \brief Contains functors for calculating vertical diffusivity and viscosity
+///
+/// This header defines functors to be called by the time-stepping scheme
+/// to calculate the vertical diffusivity and viscosity
+//
+//===----------------------------------------------------------------------===//
+
+#include "AuxiliaryState.h"
+#include "Config.h"
+#include "HorzMesh.h"
+#include "MachEnv.h"
+#include "OmegaKokkos.h"
+#include "TimeMgr.h"
+#include "VertCoord.h"
+#include <string>
+
+namespace OMEGA {
+
+enum class VertMixType {
+   PP, /// Pacanowski and Philander (1981)
+   KPP /// K-Profile Parameterization (Large et al., 1994) *** not implemented
+       /// yet ***
+};
+
+class ConvectiveMix {
+ public:
+   bool Enabled = true; ///< Enable convective mixing flag
+
+   // Convective mixing parameters
+   Real ConvDiff =
```

Comment:
I realize that these (and similar parameters for other options) will be overwritten by config values, but might want to initialize to zero or some unphysical value to cause it to fail or produce bad results if not initialized later.